components/authz-service/storage/postgres/migration/sql/47_staged_rules_tables.up.sql

```diff
@@ -17,41 +17,56 @@ CREATE TABLE iam_staged_rule_conditions (
   value TEXT[] NOT NULL,
   attribute TEXT NOT NULL,
   operator TEXT NOT NULL,
+  deleted BOOLEAN,
   updated_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
   created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP
 );
 
-CREATE FUNCTION valid_staged_rule_state_change(
-  old_state staged_rule_state, new_state staged_rule_state
-) RETURNS boolean
-LANGUAGE sql AS
-$$
-  SELECT EXISTS(
-    SELECT 1
-    FROM (VALUES
-      ('updated', 'deleted'),
-      ('updated', 'updated')
-    ) AS valid(old, new)
-    WHERE valid.old::staged_rule_state=old_state
-    AND valid.new::staged_rule_state=new_state
+CREATE OR REPLACE FUNCTION
+  projects_match_for_staged_rule(_project_id TEXT, _projects_filter TEXT[])
```

Comment:
i don't think we need this function, we could re-use the old `projects_match_for_rule`. unless i'm missing how this one is different 🤔

Reply:
we can re-use the old one-- see my note to Tyler, though, for other consideration 🤷‍♀